homework5/app.js

```diff
@@ -0,0 +1,189 @@
+// 1. Создать две функции и дать им осмысленные названия:
+// - первая функция принимает массив и колбэк (одна для всех вызовов)
+// - вторая функция (колбэк) обрабатывает каждый элемент массива (для каждого вызова свой
+// callback)
+// Первая функция возвращает строку “New value: ” и результат обработки:
+// firstFunc([‘my’, ‘name’, ‘is’, ‘Trinity’], handler1) → “New value: MyNameIsTrinity” \
+// firstFunc([10, 20, 30], handler2) → “New value: 100, 200, 300,” 
+// firstFunc([{age: 45, name: ‘Jhon’}, {age: 20, name: ‘Aaron’}], handler3) → “New value: Jhon is 45, Aaron is 20,”
+// firstFunc([‘abc’, ‘123’], handler4) → “New value: cba, 321,” // строки инвертируются
+// Подсказка: secondFunc должна быть представлена функцией, которая принимает один аргумент (каждый элемент массива) и возвращает результат его обработки
+
+let arrayOfData1 = ['my', 'name', 'is', 'Trinity'];
+let arrayOfData2 = [10, 20, 30];
+let arrayOfData3 = [{age: 45, name: 'Jhon'}, {age: 20, name: 'Aaron'}];
+let arrayOfData4 = ['abc', '123'];
+
+function createStringFromArray (array, callback) {
+    let string = '';
+
+    for (index in array) {
```

Comment:
it's unusual. Using `for in` for array. Usually `for in` is used for Object which got some unique propName. For array it's props will be just 0,1,2,3,4,5. As you can see here. It's better to use here just `for`